### ISSUE TITLE
[3.x] with gles3 html5 export, 3d scene is not rendered

### ISSUE THREAD

**Issue by uuuuuup** [godotengine/godot#50239]
### Godot version

3.x latest 153b67419f640830dfa4355a43178f2cc71540bb

### System information

UBUNTU, GLES3, nVidia 1080ti

### Issue description

With gles3 html5 export, 3d scene is not rendered.
With gles2 html5 export, it is okay.
GLES3 is not support any more in HTML5 export?

### Steps to reproduce

1. select render mode to GLES3
2. export HTML5

[webtest.zip](https://github.com/godotengine/godot/files/6778110/webtest.zip)

**Comment by clayjohn** [comment]
Can you please provide more information? Are you getting any errors? Do you see anything at all?

**Comment by uuuuuup** [comment]
I build godot engine 3.x latest and html5 export template.
I created a new project, added cube, and run in html5.
The sky is seen, but cube is not seen.

**Comment by Calinou** [comment]
@uuuuuup Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

Also, the GLES3 renderer has many known issues on HTML5. I would recommend to sticking to GLES2 for performance and compatibility reasons.

**Comment by uuuuuup** [comment]
[webtest.tar.gz](https://github.com/godotengine/godot/files/6778085/webtest.tar.gz)

I upload minimal project.

**Comment by uuuuuup** [comment]
Hi @BastiaanOlij 

I found the reason of this issue.
This is caused by commit-id : 8f8c9c2f5767bd6270bb28ce9f5beab27ccf23f9
If I change as below, it is fixed.

```
--- a/drivers/gles3/rasterizer_scene_gles3.h
+++ b/drivers/gles3/rasterizer_scene_gles3.h
@@ -157,7 +157,7 @@ public:
                        uint32_t view_index;
 
                        // make sure this struct is padded to be a multiple of 16 bytes for webgl
-                       float pad[1];
+                       float pad[3];
 
                } ubo_data;
```

Could you please check it?

**Comment by akien-mga** [comment]
>  This is caused by commit-id : 8f8c9c2

That's a bit surprising as that commit adds 4 bytes (`uint32_t`) and removes 4 bytes (one padding `float`) so it should be the same as before. But indeed the sum (before and after that commit) is 148 bytes which needs 12 bytes to pad to a multiple of 16, so the proposed fix seems correct. Feel free to submit a PR :)

**Comment by akien-mga** [comment]
Maybe you can also add a `static_assert` to make sure this doesn't break again:
```
diff --git a/drivers/gles3/rasterizer_scene_gles3.h b/drivers/gles3/rasterizer_scene_gles3.h
index ddfa47154e..25872ea7a0 100644
--- a/drivers/gles3/rasterizer_scene_gles3.h
+++ b/drivers/gles3/rasterizer_scene_gles3.h
@@ -157,9 +157,9 @@ public:
                        uint32_t view_index;
 
                        // make sure this struct is padded to be a multiple of 16 bytes for webgl
-                       float pad[1];
-
+                       float pad[3];
                } ubo_data;
+               static_assert(sizeof(SceneDataUBO) % 16 == 0);
 
                GLuint scene_ubo;
 
```

**Comment by akien-mga** [comment]
Fixed by #50276.
